pkg/meshsync/cluster/pipeline/pipeline.go

```diff
@@ -15,13 +15,26 @@ var (
 		Concurrent: true,
 		Steps:      []pipeline.Step{},
 	}
+
+	LocalDiscoveryStage = &pipeline.Stage{
+		Name:       "Local-Resource-Discovery",
+		Concurrent: true,
+		Steps:      []pipeline.Step{},
+	}
 )
 
 func (cluster *Cluster) InitializePipeline() (*pipeline.Pipeline, error) {
 
-	// Mesh Discovery Stage
+	// Global discovery
 	gdstage := GlobalDiscoveryStage
 	gdstage.AddStep(cluster)
```

Comment:
@kumarabd, Why are we adding cluster as a step here?

Reply:
Yes ideally should not be in there, must be a separate pipeline. Can you make that change? @kalradev

Reply:
Should I remove it? or make a new stage for it?